### PR TITLE
Cookie jwt

### PR DESCRIPTION
**Pull request by julienmachon**
## What is this?
This is a POC to show how the JWT can be store in a cookie, rather than local storage and therefor be used on the server for SSR.

## How does it work?:

The express app (the server) has a `/authSuccess` route setup which will setup the client cookie with the auth token. The idea is that the OAuth provider will be given `/authSucess` as a redirect URL

When the server receives a request, it initialises the AuthContext value to `authenticated: true` if the client has a valid `nexusAuth` cookie, or `authenticated: false` is no valid cookie is present.

The client read its own cookies and then setups up the AuthContext the same way the server does.

Finally, in the Router, the `PrivateRoute` component checks if the user is authenticated from the AuthContext and redirects to the `/login` page if not.

So basically, from the app point of view, the flow is:

```
authData;
if (client) authData = setAuthFromLocalCookie()
if(server) authData = setAuthtFromRequestCookie()

renderAppWithAuthContext(authData)
```

### CODE REVIEW

**Comment by mdupont** [comment]
Please also see https://infosec.mozilla.org/guidelines/web_security.html#cookies and https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet

File: src/client/index.tsx [hunk 1]
```diff
@@ -2,18 +2,48 @@ import React = require('react');
...
+  <AuthContext.Provider value={getAuthData()}>
+    <BrowserRouter basename={base}>
+      <App />
+    </BrowserRouter>
+  </AuthContext.Provider>,
   document.getElementById('app'),
 );
 
+// DEVELOPMENT ONLY
+// if Hot module Replacement is enables
```
Comment (mdupont) [inline src/client/index.tsx]: typo

**Comment by julienmachon** [comment]
Security options will be addressed in BlueBrain/nexus#228
